complex/ft_test.c

```diff
@@ -205,15 +218,10 @@ static void ft_format_iov_random(struct iovec *iov, size_t cnt, char *buf,
 	offset = 0;
 	for (i = 0; i < cnt; i++) {
 		/* If last IOV then use remaining data. */
-		if (i == (cnt - 1)) {
+		if (i == (cnt - 1))
 			weight = len;
-		} else {
-			/* Get a weight between 1 and the remaining length minus
-			 * the remaining IOV count. This is so we can reserve at
-			 * least a length of 1 for every IOV.
-			 */
-			weight = (rand() % (len - (cnt - i))) + 1;
-		}
```

Comment:
That should be avoided, so providers don't need to check for it.  At least until support is standardized through the API.